java/src/jmri/jmrix/can/cbus/CbusReporter.java

```diff
@@ -25,40 +24,81 @@
  * A PARTICULAR PURPOSE. See the GNU General Public License for more details.
  * <p>
  *
+ * CBUS Reporters can accept 
+ * 5-byte unique Classic RFID on DDES or ACDAT OPCs,
+ * CANRC522 / CANRCOM DDES OPCs.
+ * 
  * @author Mark Riddoch Copyright (C) 2015
- * @author Steve Young Copyright (c) 2019
+ * @author Steve Young Copyright (c) 2019, 2020
  *
  */
-public class CbusReporter extends AbstractReporter implements CanListener {
+public class CbusReporter extends AbstractRailComReporter implements CanListener {
```

Comment:
A design question/suggestion triggered by this change in base class....

I see later on in this revised class that you have 3 different types of tags ( RFID, RC522, and RailComm ) you might be handling here,  suggesting there are three different types of hardware involved.

If that is the case, this class as modified violates the Single Responsibility Principle because it has three reasons it could change.  For long term maintenance and testing reasons, It is advisable to have a different implementation for each of the three types.  That also simplified the logic, because each class has to handle only the information pertaining to the hardware it handles.

For an example of where we do this elsewhere, see the Z21 Reporter implementation.

Reply:
Hi Paul, thanks for taking the time to look over, it is appreciated.

I do hear what you are saying about Single Responsibility.
Eg, there's a high chance that CANRCOM will be able to implement the CollectingReporter interface, any code relying on instanceof CollectingReporter will show incorrect results for the other types.

Inspired by Z21ReporterManager, I'm thinking that the next step in development of this would be for "describing" system names, eg.
MR123 - Classic RFID on device 123 - extends AbstractIdTagManager
MR[CANRC522]123 - CANRC522 on device 123 - extends AbstractRailComReporter 
MR[CANRCOM]123 - CANRCOM on device 123 - extends AbstractRailComReporter implements CollectingReporter 

As the OPC formats themselves are mid-development ( and I still need to add actual code for CANRCOM! ), there are likely to be a decent number of updates surrounding this, so I'll look at refactoring to unique classes soon.